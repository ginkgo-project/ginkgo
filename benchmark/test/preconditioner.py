#!/usr/bin/env python3
import test_framework

# check that all input modes work:
# parameter
test_framework.compare_output(
    ["-input", '[{"size": 100, "stencil": "7pt"}]'],
    expected_stdout="preconditioner.simple.stdout",
    expected_stderr="preconditioner.simple.stderr",
)

# stdin
test_framework.compare_output(
    [],
    expected_stdout="preconditioner.simple.stdout",
    expected_stderr="preconditioner.simple.stderr",
    stdin='[{"size": 100, "stencil": "7pt"}]',
)

# input file
test_framework.compare_output(
    ["-input", str(test_framework.sourcepath / "input.mtx.json")],
    expected_stdout="preconditioner.simple.stdout",
    expected_stderr="preconditioner.simple.stderr",
)

# profiler annotations
test_framework.compare_output(
    [
        "-input",
        '[{"size": 100, "stencil": "7pt"}]',
        "-profile",
        "-profiler_hook",
        "debug",
    ],
    expected_stdout="preconditioner.profile.stdout",
    expected_stderr="preconditioner.profile.stderr",
)
