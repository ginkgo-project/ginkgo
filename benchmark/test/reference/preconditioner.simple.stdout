
[
    {
        "size": 125,
        "stencil": "7pt",
        "preconditioner": {
            "none": {
                "generate": {
                    "components": {
                        "generate(<typename>)": 1.0,
                        "overhead": 1.0
                    },
                    "time": 1.0,
                    "repetitions": 10
                },
                "apply": {
                    "components": {
                        "apply(<typename>)": 1.0,
                        "copy(<typename>)": 1.0,
                        "dense::copy": 1.0,
                        "overhead": 1.0
                    },
                    "time": 1.0,
                    "repetitions": 10
                },
                "completed": true
            }
        }
    }
]
